app/controllers/trades_controller.rb

```diff
@@ -9,6 +11,10 @@ def new
     @trade = Trade.new
   end
 
+  def show
+    @trade = Trade.find(params[:id])
```

Comment:
You can move this to an before_action callback as you will need to use it too for the update